python/tritonserver/_api/_response.py

```diff
@@ -29,293 +29,37 @@
 from __future__ import annotations
 
 import asyncio
+import concurrent
 import inspect
 import queue
 from dataclasses import asdict, dataclass, field
 from typing import TYPE_CHECKING, Optional
 
+import numpy
 from tritonserver._api import _model
 
 if TYPE_CHECKING:
     from tritonserver._api._model import Model
 
+from tritonserver._api._dlpack import DLDeviceType as DLDeviceType
 from tritonserver._api._logging import LogMessage
+from tritonserver._api._memorybuffer import MemoryBuffer
 from tritonserver._api._tensor import Tensor
 from tritonserver._c.triton_bindings import (
     InternalError,
+    InvalidArgumentError,
     TritonError,
-    TRITONSERVER_InferenceRequest,
 )
 from tritonserver._c.triton_bindings import TRITONSERVER_LogLevel as LogLevel
+from tritonserver._c.triton_bindings import TRITONSERVER_MemoryType as MemoryType
 from tritonserver._c.triton_bindings import (
     TRITONSERVER_ResponseCompleteFlag,
     TRITONSERVER_Server,
 )
 
-
-class AsyncResponseIterator:
-
-    """Asyncio compatible response iterator
-
-    Response iterators are returned from model inference methods and
-    allow users to process inference responses in the order they were
-    received for a request.
-
-    """
-
-    def __init__(
-        self,
-        model: _model.Model,
-        request: TRITONSERVER_InferenceRequest,
-        user_queue: Optional[asyncio.Queue] = None,
-        raise_on_error: bool = False,
-        loop: Optional[asyncio.AbstractEventLoop] = None,
-    ) -> None:
-        """Initialize AsyncResponseIterator
-
-        AsyncResponseIterator objects are obtained from Model inference
-        methods and not instantiated directly. See `Model` documentation.
-
-        Parameters
-        ----------
-        model : Model
-            model associated with inference request
-        request : TRITONSERVER_InferenceRequest
-            Underlying C binding TRITONSERVER_InferenceRequest
-            object. Private.
-        user_queue : Optional[asyncio.Queue]
-            Optional user queue for responses in addition to internal
-            iterator queue.
-        raise_on_error : bool
-            if True response errors will be raised as exceptions.
-        loop : Optional[asyncio.AbstractEventLoop]
-            asyncio loop object
-
-        """
-
-        if loop is None:
-            loop = asyncio.get_running_loop()
-        self._loop = loop
-        self._queue = asyncio.Queue()
-        self._user_queue = user_queue
-        self._complete = False
-        self._request = request
-        self._model = model
-        self._raise_on_error = raise_on_error
```

Comment:
are we missing raise_on_error with new iterator?

Reply:
yes, added raise on error: [Add raise on error to response iterators](https://github.com/triton-inference-server/core/pull/426/commits/898ef02f33e74822a6b175068f4ae7cdf13c62d4)